includes/class-wc-calypso-bridge-admin-menu.php

```diff
@@ -10,11 +10,331 @@
  */
 class Ecommerce_Atomic_Admin_Menu extends \Automattic\Jetpack\Dashboard_Customizations\Atomic_Admin_Menu {
 
+	const WPCOM_ECOMMERCE_MANAGED_PAGES = array(
+		'wc-admin',
+		'wc-admin&path=/customers',
+		'edit.php?post_type=shop_order',
+		'wc-reports',
+		'wc-settings',
+		'wc-status',
+		'wc-addons',
+	);
+
+	/**
+	 * Override constructor and add custom actions.
+	 */
+	public function __construct() {
+		parent::__construct();
+		add_action( 'admin_menu', array( $this, 'add_woocommerce_menu' ), 99999 );
+		add_filter( 'menu_order', array( $this, 'menu_order' ), 100 );
+
+		if ( ! $this->is_api_request ) {
+			add_filter( 'submenu_file', array( $this, 'modify_woocommerce_menu_highlighting' ), 99999 );
+		}
+
+		// Move Orders.
+		// TODO: What about the COT menu?
+		add_filter( 'woocommerce_register_post_type_shop_order', function( $args ) {
+			$args[ 'labels' ][ 'add_new' ] = __( 'Add new', 'woocommerce' );
+			$args[ 'show_in_menu' ]        = true;
+			$args[ 'menu_icon' ]           = 'dashicons-cart';
+			return $args;
+		} );
+	}
+
+	/**
+	 * Groups WooCommerce items.
+	 */
+	public static function menu_order( $menu_order ) {
+
+		// Initialize our custom order array.
+		$woocommerce_menu_order = array();
+
+		// Get the index of our managed pages.
+		$orders    = array_search( 'edit.php?post_type=shop_order', $menu_order, true );
+		$customers = array_search( 'admin.php?page=wc-admin&path=/customers', $menu_order, true );
+		$products  = array_search( 'edit.php?post_type=product', $menu_order, true );
+		$analytics = array_search( 'wc-admin&path=/analytics/overview', $menu_order, true );
+		$marketing = array_search( 'woocommerce-marketing', $menu_order, true );
+		$extensions                = array_search( 'woocommerce', $menu_order, true );
+		$woocommerce_separator     = array_search( 'separator-woocommerce', $menu_order, true );
+		$woocommerce_separator_top = array_search( 'wc-calypso-bridge-separator-top', $menu_order, true );
+		$woocommerce_payments      = array_search( 'wc-admin&path=/payments/connect', $menu_order, true );
```

Comment:
Minor thought: might it make sense to name these variables more explicitly with something like a `_position` or `_index` suffix?